Public org page

richardolsson opened the pull request:
## Description
This PR creates an MVP version of public org pages.

## Screenshots
### Including events from sub-organization
![image](https://github.com/user-attachments/assets/7cfbb111-643d-4c5c-b898-ac651458fee9)

### Signed up for event
![image](https://github.com/user-attachments/assets/1dbaca9a-e8a8-452d-8a75-6fc67b4ee595)

## Changes
* Adds a home page for organizations that lists their events ("Calendar" tab)
* Adds a page that lists sub-organizations ("Explore")
* Implements MVP flow for logging in to sign up for an event

## Notes to reviewer

## Related issues
This solves most of #1618, and (I believe) all the parts that we know are relevant. So IMO this closes #1618.

ziggabyte commented on src/features/organizations/l10n/messageIds.ts:
```diff
@@ -30,4 +46,16 @@ export default makeMessages('feat.organizations', {
+    }>(
+      'There are {eventsElem} {numEvents, plural, =1 {additional event} other {additional events}} in {orgsElem} {numOrgs, plural, =1 {suborganization} other {suborganizations}}.'
+    ),
+    headline: m('Dig deeper'),
```
I really like this informal tone!

ziggabyte reviewed (approved):
Noticed when playing in the preview build. Nothing feels super crucial, so I'm gonna approve, but build some of it if you feel like it! 

- The height of the header changes slightly when the button to go "upstairs" is visible: 
![bild](https://github.com/user-attachments/assets/7e946ece-5ce4-42c8-b410-299826024515)

- I get the "text" cursor when hovering this little chip, i don't super love it since it's inside a chip and it doesn't feel like text but more like a "thing" than a text? i don't know if that makes sense, it's just vibes
![bild](https://github.com/user-attachments/
assets/4bbb995c-2b48-4098-816e-6219fd8aeeda)

- For some reason the "My other organization" appears 2 times in the "Explore" list after I've clicked around between orgs for a bit. Did not notice the exact moment it got added twice, but I noticed it after i had played with just going in and out of different orgs for a bit. Then after clicking in and out of orgs again for a while, suddenly there was only one "My other org" again: 
![bild](https://github.com/user-attachments/assets/2aec6d4c-8833-4d4c-952e-79c73f8839b9)

- This is a design thing, but after i clicked the "dig deeper" banner to get more events, I find myself longing to be able to visually distinguish between events from the org i am in, and from other orgs. Now the only way to notice is in the gray name of the org on the event card, which is not super visible. 
![bild](https://github.com/user-attachments/assets/51c2f242-4334-403d-9a3a-ac79ce211b31)

- Not a breaking thing, but it feels a bit odd to arrive at an empty "Explore" tab. Could this have an empty state to explain why it's empty maybe? 
![bild](https://github.com/user-attachments/assets/a4280d98-2db5-4cf2-89dc-1264ef56b472)

- Same here with the empty calendar, it feels a bit odd that there is no empty state to explain why it's empty
![bild](https://github.com/user-attachments/assets/7626911c-26e6-496e-ad1e-2d23c0f08894)

- General feedback: the same as I said in k&k, it feels quite a lot more "stiff" than the gen2 activist portal, which I am not sure is the conscious intention.

richardolsson commented:
Thank you @ziggabyte for reviewing! Below are my answers to the points you've raised, and I will request a new review when I've fixed the ones I agree need fixing.

> * The height of the header changes slightly when the button to go "upstairs" is visible:

I will see if I can do something about this!

> * I get the "text" cursor when hovering this little chip, i don't super love it since it's inside a chip and it doesn't feel like text but more like a "thing" than a text? i don't know if that makes sense, it's just vibes

Should be a simple fix, so I will have a look.

> * For some reason the "My other organization" appears 2 times in the "Explore" list after I've clicked around between orgs for a bit. Did not notice the exact moment it got added twice, but I noticed it after i had played with just going in and out of different orgs for a bit. Then after clicking in and out of orgs again for a while, suddenly there was only one "My other org" again:

I did see this before, but then I thought I had fixed it. I will investigate further.

> * This is a design thing, but after i clicked the "dig deeper" banner to get more events, I find myself longing to be able to visually distinguish between events from the org i am in, and from other orgs. Now the only way to notice is in the gray name of the org on the event card, which is not super visible.

This is actually a somewhat conscious design choice. I understand the longing, but I'm just not sure it's something a normal user (who is not mainly testing a sub-org feature, but actually looking for events) care about whether it's the parent org or the sub-org that's organizing. They do probably care about geography though, but I'm not sure we can solve that any other way than using a map.

> * Not a breaking thing, but it feels a bit odd to arrive at an empty "Explore" tab. Could this have an empty state to explain why it's empty maybe?

> * Same here with the empty calendar, it feels a bit odd that there is no empty state to explain why it's empty

I will add the empty states that @vbw designed. 

> * General feedback: the same as I said in k&k, it feels quite a lot more "stiff" than the gen2 activist portal, which I am not sure is the conscious intention.

I agree, and I think we should spend some designer time on activist portal look and feel!

richardolsson commented:
Ok I believe I have changed all the things that requires changing now.

The only thing I didn't fix was an empty state for the "Explore" tab. Instead, I made sure to hide that tab when there is nothing there. It's technically still possible to find the page, but not from any visible links.

Wanna review again, @ziggabyte ?

ziggabyte reviewed (commented):
I don't know if this is intentional, but i get events from My fourth org when I'm in My third org. It feels like it might be a mistake? 
![bild](https://github.com/user-attachments/assets/1b886cc0-df64-48a1-9f55-d0f586efaf81)

richardolsson commented:
> I don't know if this is intentional, but i get events from My fourth org when I'm in My third org. It feels like it might be a mistake? 
> ![bild](https://github.com/user-attachments/assets/1b886cc0-df64-48a1-9f55-d0f586efaf81)
> 

It's not a mistake, but I understand how it can seem that way when you're looking with a critical eye.

Third org doesn't have any events of its own, so it displays events from its sub orgs immediately, without need for user interaction, instead of displaying an empty state.

ziggabyte reviewed (approved):
This feels like a really really nice step into the full version of the activist portal!! :D
